src/docs/guides/django.md

```diff
@@ -0,0 +1,254 @@
+
+```bash
+python -m pip install gunicorn whitenoise psycopg[binary,pool]
+```
+
+[whitenoise](https://whitenoise.readthedocs.io/en/stable/index.html) is a Python package for serving static files directly from your web app. It serves compressed content and sets far-future cache headers on content that won't change.
+
+[gunicorn](https://gunicorn.org) is a production ready web server.
+
+[pyscog](https://www.psycopg.org/psycopg3/docs) is python package that allows Django work with Postgresql.
+
+2. Import the `os` module:
+
+Open the `liftoff/settings.py` file located in the inner `liftoff` directory (the one containing the main project settings).
+
+At the top of the file, add the following line to import the `os` module, placing it just before the `Path` import:
+
+```python
+import os
+from pathlib import Path
+```
+
+3. Configure the database:
+
+A fresh Django project uses SQLite by default, but we need to switch to PostgreSQL. 
+
+Open the `liftoff/settings.py` file. In the Database section, replace the existing configuration with:
+
+```python
+# Database
+# https://docs.djangoproject.com/en/5.1/ref/settings/#databases
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.postgresql',
+        'NAME': os.environ["PGDATABASE"],
+        'USER': os.environ["PGUSER"],
+        'PASSWORD': os.environ["PGPASSWORD"],
+        'HOST': os.environ["PGHOST"],
+        'PORT': os.environ["PGPORT"],
+    }
+}
+```
+
+4. Static files configuration:
+
+We'll configure Django to serve static files using [WhiteNoise](https://whitenoise.readthedocs.io/en/stable/index.html). 
+
+Open `liftoff/settings.py` and configure the static files settings:
+
+```python
+STATIC_URL = 'static/'
+
+STATIC_ROOT = os.path.join(BASE_DIR, "staticfiles")
+STATICFILES_DIRS = [os.path.join(BASE_DIR, "static")]
+```
+
+Add the WhiteNoise middleware in the **MIDDLEWARE** section, just below the [security middleware](https://docs.djangoproject.com/en/5.1/ref/middleware/#module-django.middleware.security):
+
+```python
+MIDDLEWARE = [
+    'django.middleware.security.SecurityMiddleware',
+    'whitenoise.middleware.WhiteNoiseMiddleware',
+    'django.contrib.sessions.middleware.SessionMiddleware',
+    'django.middleware.common.CommonMiddleware',
+    'django.middleware.csrf.CsrfViewMiddleware',
+    'django.contrib.auth.middleware.AuthenticationMiddleware',
+    'django.contrib.messages.middleware.MessageMiddleware',
+    'django.middleware.clickjacking.XFrameOptionsMiddleware',
+]
+```
+
+5. Update `ALLOWED_HOSTS` settings:
+
+```python
+ALLOWED_HOSTS = ["*"]
+```
+
+This setting represents the list of all the host/domain names our Django project can serve.
+
+6. Create a **static** folder:
+
+Inside your `liftoff` directory, create a static folder where all static assets will reside.
+
+7. Create a `requirements.txt` file: 
+ 
+To track all the dependencies for deployment, create a `requirements.txt` file:
+
+```bash
+pip freeze > requirements.txt
```

Comment:
maybe mention that this is only safe to run since we are in a venv, otherwise this will freeze all packages the user has ever installed on their system

Reply:
Oh this is why I started the venv because I knew we will get to this point. I'll add that as a note.

Reply:
Added the note.